api/base/serializers.py

```diff
@@ -225,7 +225,7 @@ def to_internal_value(self, data):
             if request.method in utils.UPDATE_METHODS and not utils.is_bulk_request(request):
                 id_field = getattr(self.root.instance, self.source, '_id')
                 if id_field != data:
-                    raise Conflict()
+                    raise Conflict(detail=('The resource id you specified "' + data + '" does not match the id of the resource you specified "' + id_field + '".'))
```

Comment:
You'll want to verify I'm correct about the second sentence though. I'm pretty sure I am, but if this error hits before the 404, that would be bad advice.